### PR TITLE
128: as a user i want to see file changes

### PR DESCRIPTION
**Pull request by IlliaLop**

### CODE REVIEW

File: backend/Squirrel.Core/Squirrel.Core.BLL/Services/CommitChangesService.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        _httpClientService = httpClientService;
+        _userIdGetter = userIdGetter;
+        _configuration = configuration;
+    }
+    
+    public async Task<List<DatabaseItemContentCompare>> GetContentDiffsAsync(int commitId, Guid tempBlobId)
+    {
+        return await _httpClientService.GetAsync<List<DatabaseItemContentCompare>>(
+            $"{_configuration["SqlServiceUrl"]}/api/ContentDifference/{commitId}/{tempBlobId}");
```
Comment (OlegKarapysh) [inline backend/Squirrel.Core/Squirrel.Core.BLL/Services/CommitChangesService.cs]: remove magic values

File: frontend/src/app/core/services/commit-changes.service.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+@Injectable({
+    providedIn: 'root',
+})
+export class CommitChangesService {
+    private readonly commitChangesRoute = '/api/commitchanges';
+
+    private contentChangesSubject = new BehaviorSubject<DatabaseItemContentCompare[]>([]);
+
+    contentChanges$ = this.contentChangesSubject.asObservable();
```
Comment (OlegKarapysh) [inline frontend/src/app/core/services/commit-changes.service.ts]: access modifier is missing here

**Comment by IlliaLop** [comment]
image

**Comment by serhiiyasenev** [comment]
can we use lighter/transparent colors (like in GitHub for example)?
image

File: frontend/src/app/shared/components/code/code.component.ts [hunk 3]
```diff
@@ -24,25 +24,31 @@ export class CodeComponent implements OnChanges {
 
     ngOnChanges(changes: SimpleChanges): void {
         if (changes['textPair']) {
+            const codeTable = this.el.nativeElement.querySelector('#code-table');
+
+            codeTable.innerHTML = ''; // Очищаємо вміст перед вставкою нового тексту
```
Comment (serhiiyasenev) [inline frontend/src/app/shared/components/code/code.component.ts]: `// Очищаємо вміст перед вставкою нового тексту`

in English or remove

**Comment by IlliaLop** [comment]
> can we use lighter/transparent colors (like in GitHub for example)? image

I think these colors that are in the code component are also normal

File: backend/Squirrel.Core/Squirrel.Core.BLL/Services/BranchService.cs [hunk 4]
```diff
@@ -32,6 +33,21 @@ public async Task<BranchDto> AddBranchAsync(int projectId, BranchCreateDto branc
         return _mapper.Map<BranchDto>(createdBranch);
     }
 
+    public async Task<int> GetLastBranchCommit(int branchId)
```
Comment (OlegKarapysh) [inline backend/Squirrel.Core/Squirrel.Core.BLL/Services/BranchService.cs]: Async suffix

File: frontend/src/app/core/services/commit-changes.service.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    providedIn: 'root',
+})
+export class CommitChangesService {
+    private readonly commitChangesRoute = '/api/commitchanges';
+
+    private contentChangesSubject = new BehaviorSubject<DatabaseItemContentCompare[]>([]);
+
+    public contentChanges$ = this.contentChangesSubject.asObservable();
+
+    // eslint-disable-next-line no-empty-function
```
Comment (OlegKarapysh) [inline frontend/src/app/core/services/commit-changes.service.ts]: remove comment

**Review by serhiiyasenev — DISMISSED** [review]

**Comment by Anatolii-Ivanov** [comment]
Remove json Object please, just show definition, as it makes no sense for user

**Comment by Anatolii-Ivanov** [comment]
> > can we use lighter/transparent colors (like in GitHub for example)? image
> 
> I think these colors that are in the code component are also normal

Please, update them)
There are too much

**Comment by IlliaLop** [comment]
> > > can we use lighter/transparent colors (like in GitHub for example)? image
> > 
> > 
> > I think these colors that are in the code component are also normal
> 
> Please, update them) There are too much

image

**Review by Anatolii-Ivanov — APPROVED** [review]

**Review by serhiiyasenev — APPROVED** [review]

**Pull request merged**